### DISCUSSION TITLE
i18n via message ids

### DISCUSSION THREAD

**Discussion by alrusdi** [bafolts/terraforming-mars#2882] [Localization]
We have a lot of problems with consistency, support and testability of current i18n subsystem.

And general cause of all of this problems is lack of "official" source strings to translate.

@kberg suggested a way to solve it:
1) We create an English to English translation where keys (source string) is very specific message ids and value (translation) is actual English text to be displayed to end user. Translation can look like this:
{"MSG00001": "Cards list"}
2) For all translations it's required to  use that English to English translation as reference like this:
{"MSG00001": "Список карт"}

As developers, when we want to show some text to user in browser we must use that message ids instead of actual text.
Code will look something like this:

```<a ... v-i18n>MSG00001</a>```
or
```
...
      metadata: {
        description: 'MSG00321',
...
```
According to my experience it is very uncomfortable/annoying for developers, but it works great in terms of consistency, support and testability.

What do you think about it? Is it reasonable to you? Do we have a better options?

**Comment by vsrisuknimit** [comment]
I would not want to count the message id one by one across all cards.

Instead of 'MSG00321', can we go by card id '134_description', '134_action_1' or card name 'ExtremeColdFungus_description', 'ExtremeColdFungus_action1_1'?

Is there also a way to have the English translation in the same place as the cards for readability of the code?

**Reply by alrusdi** [reply]
Yes, id format is a subject of discussion too.

**Reply by alrusdi** [reply]
I don't sure about English translation. What do you mean? Combined "message_id" + "english_text"?

**Comment by bafolts** [comment]
Using ids for the strings we translate makes sense to me. We don't have to update every string in one commit so we could roll out the change slowly. If when running the `t` function we found the key we could use the value, otherwise we would revert to what we have today. With any system in place any developer is going to be able to introduce a string that we forget to translate so the old system being in place simultaneously can help spot the places we still need to update to use a key.

These ids would also help us lower the size of the `/api/game/logs` route. Each log message would become the message id instead of the full message string.
